akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -841,6 +853,22 @@ private[akka] class ActorSystemImpl(
     new Settings(classLoader, config, name, setup)
   }
 
+  private var _licenseKeyExpiry: Option[LocalDate] = None
+
+  override def licenseKeyExpiry: Option[LocalDate] = _licenseKeyExpiry
+
+  override def getLicenseKeyExpiry: Optional[LocalDate] = _licenseKeyExpiry.toJava
+
+  private def setLicenseKeyExpiry(expiry: LocalDate): Unit = {
+    _licenseKeyExpiry match {
+      case None =>
+        _licenseKeyExpiry = Some(expiry)
+      case Some(exp) =>
+        if (expiry.isBefore(exp))
+          _licenseKeyExpiry = Some(expiry)
+    }
+  }
```

Comment:
Will probably be read by the same thread starting the system but, synchronized/volatile something for good measure?

Reply:
ok, probably plenty of other visibility boundaries in the startup, but doesn't hurt